File: ApplicationServer/ManhwaTracker/ManhwaTrackerApplicationServer/Services/ManhwaService.cs
using ManhwaTrackerApplicationServer.Models.Manhwa;
using ManhwaTrackerApplicationServer.Repositories;

namespace ManhwaTrackerApplicationServer.Services;

public class ManhwaService : IManhwaService
{
    private readonly IManhwaRepository _manhwaRepository;

    public ManhwaService(IManhwaRepository manhwaRepository)
    {
        _manhwaRepository = manhwaRepository;
    }
    public async Task<IEnumerable<Manhwa>> GetAllAsync()
    {
        return await _manhwaRepository.GetAllAsync();
    }

    public async Task<Manhwa> GetByIdAsync(int id)
    {
        return await _manhwaRepository.GetByIdAsync(id);
    }

    public async Task<IEnumerable<Manhwa>> GetByTitleAsync(string title)
    {
        return await _manhwaRepository.GetByTitleAsync(title);
    }
}

File: ReactManhwaTracker/manhwa-tracker/src/data/manhwa/manhwa-service.ts
import Manhwa from "../../models/manhwa/manhwa";

export interface ManhwaService {
    getAllManhwas: () => Promise<Manhwa[]>;
    getManhwasByTitle: (title: string) => Promise<Manhwa[]>;
    getManhwaById: (id: number) => Promise<Manhwa>;
}
